### PR TITLE
Added signup page

### LINKED ISSUE #49: Sign up page
Add a signup page that saves the profile to the DB

### PR DESCRIPTION
**Pull request by San7o**
Closes #49

### CODE REVIEW

File: src/main/java/Signup.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+      // DB Connection
+      try {
+
+        // Connecting
+        String url = "jdbc:derby://localhost:1527/DerbyDB";
+        Connection con = DriverManager.getConnection(url);
+
+        // Get parameters 
+        String nome = req.getParameter("nome");
```
Comment (Qw3rty26) [inline src/main/java/Signup.java:55]: Shouldn't these be made using a java bean?

File: src/main/java/Signup.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        // Check if the user already exists
+        if (ProfiloDAO.UserExists(username, con)) {
+
+            req.setAttribute("error", "Username gia esistente");
+            req.getRequestDispatcher("/error").forward(req, res);
+        }
+
+
+        //Create Java Bean for profile
+        ProfiloBean profiloBean = new ProfiloBean(nome,
```
Comment (Qw3rty26) [inline src/main/java/Signup.java:87]: You cannot create String variables and then pass them to create a bean....That deletes the sole purpose of having a java bean in the first place.

**Review by Qw3rty26 — CHANGES_REQUESTED** [review]
Wonderfull code @San7o! Before merging this PR though, i added two comments on how the variables needed for the signup are actually handled. Maybe it is my missunderstanding on how a MVC compliant webapp is made. So hit me up with a response regarding my comments and ill be sure to accept your PR if everything's ok!

File: src/main/java/Signup.java (hunk 1 above)
Comment (Qw3rty26) [reply src/main/java/Signup.java:55]: By "these" i mean the "String nome", "String cognome" ecc...

File: src/main/java/Signup.java (hunk 2 above)
Comment (Qw3rty26) [reply src/main/java/Signup.java:87]: Or at least that's what i think?

File: src/main/java/Signup.java (hunk 1 above)
Comment (San7o) [reply src/main/java/Signup.java:55]: We first need to check that the parameters exist before creating the bean

**Comment by San7o** [comment]
My design decision when writing this code was the following: It's wrong and memory inefficient to create a bean with missing parameters, so first I check that the parameters exist and are valid, then I create a bean. This is debatable and maybe where is a better way. From a memory prospective, this is more efficient and from a logical prospective It's wrong to create a bean with missing parameters. On the other hand, maybe checking that the user already exists could be done with the bean instead of using only the username. My choice was to be more memory efficient. We may discuss this further.

**Comment by Qw3rty26** [comment]
I can see your point @San7o. We should wait to push this PR until we can ask _that person_ what we can and cannot do ;)

**Comment by San7o** [comment]
I had a vision, Vincenzo Marco spoke to me, he said this PR is fine and should be merged, glory to Vincenzo Marco 














(We didn't book an appointment with the messiah yet so we might just implement the feaure since It's pretty basic and maybe change it later)

**Review by Qw3rty26 — APPROVED** [review]
Very good code!!!!!

**Pull request merged**